### ISSUE TITLE
Leaderboard improvements (block input while loading, add "pause scene" option)

### ISSUE THREAD

**Issue by tristanbob** [4ian/GDevelop#3991]
## Description

Leaderboards are very simple to implement; they can even be displayed on top of existing scenes.
However, there are a few things that can make this even better.

1. While the leaderboard is loading, there is a period of time when the existing game interface is still active and responding to clicks.  This can cause unintended actions to take place. 

- [ ] Inputs to the game should be blocked while loading the leaderboard the same way they are blocked while the leaderboard is displayed.

2. While the leaderboard is loading and being displayed, the existing scene is still running.  

- [ ] Why not offer the ability to pause the active scene?  I think this would make sense as a boolean parameter to "Display leaderboard". "Pause scene while leaderboard is shown?"

@AlexandreSi

**Comment by tristanbob** [comment]
I do know that the game thinks the cursor is static while the leaderboard is up.  I discovered this because I had some text moving up and down with sine-movement and added a sound when the user hovered the cursor over the text.  

While the leaderboard was up, I noticed that the "hover" sound was playing occasionally.  I realized the game was moving the text on and off the cursor position.

**Comment by PascalLadalle** [comment]
👋 
I'm giving the leaderboard feature a try and I have a few requests/issues.
My levels are procedurally generated, so I made a leaderboard scene as explained in the wiki, and I want to switch to that scene for each level.
- I'm trying to use the leaderboard instructions using the expression "Leaderboard_"+ ToString(GlobalVariable(level)), and it's not working. Is this expression feature only compatible with the leaderboard number id? I haven't seen this mentioned, and it's quite inconvenient. 
- It would be nice to be able to duplicate leaderboards in the leaderboard editor. I mean create a new leaderboard with the same parameters as the one currently displayed, with a new ID of course, and a new or incremented name.
- The Delete pop-up closes the leaderboard editor, whether you confirm, cancel or click anywhere. Is this on purpose?

**Comment by AlexandreSi** [comment]
> * I'm trying to use the leaderboard instructions using the expression "Leaderboard_"+ ToString(GlobalVariable(level)), and it's not working. Is this expression feature only compatible with the leaderboard number id? I haven't seen this mentioned, and it's quite inconvenient.

Indeed, it only works with the leaderboard id at the moment. I see why you would be interested in such a feature.

> * It would be nice to be able to duplicate leaderboards in the leaderboard editor. I mean create a new leaderboard with the same parameters as the one currently displayed, with a new ID of course, and a new or incremented name.

That's a good idea, I'll see how/where we can put a button for that.

> * The Delete pop-up closes the leaderboard editor, whether you confirm, cancel or click anywhere. Is this on purpose?

I couldn't reproduce this bug. Could you send a video of this bug happening?
